File: clients/socket-peer.js
const socket = io("https://upgraded-barnacle-7v4xprj74jwhxr55-5000.app.github.dev/");

let clientId = null;

function registerClient(customId = null) {
    socket.emit("register", { client_id: customId });
}

socket.on("assigned_id", (data) => {
    clientId = data.client_id;
    console.log("Assigned ID:", clientId);
});

socket.on("receive_message", (data) => {
    console.log(`Message from ${data.from}: ${data.message}`);
});

socket.on("error", (data) => {
    console.log("Error:", data.message);
});

function sendMessage(targetId, message) {
    if (!clientId) {
        console.log("You are not registered.");
        return;
    }
    socket.emit("send_message", { from: clientId, target_id: targetId, message: message });
}

// Example usage:
// registerClient("user123");
// sendMessage("user456", "Hello, user456!");

File: clients/socket-peer.py
import socketio

class PeerClient:
    def __init__(self, server_url, client_id=None):
        self.sio = socketio.Client()
        self.client_id = client_id

        @self.sio.on("assigned_id")
        def on_assigned_id(data):
            self.client_id = data["client_id"]
            print(f"Assigned ID: {self.client_id}")

        @self.sio.on("receive_message")
        def on_receive_message(data):
            print(f"Message from {data['from']}: {data['message']}")

        @self.sio.on("error")
        def on_error(data):
            print(f"Error: {data['message']}")

        self.sio.connect(server_url)
        self.sio.emit("register", {"client_id": self.client_id})

    def send_message(self, target_id, message):
        if not self.client_id:
            print("You are not registered with the server.")
            return
        self.sio.emit("send_message", {"from": self.client_id, "target_id": target_id, "message": message})

    def close(self):
        self.sio.disconnect()

# Example usage
if __name__ == "__main__":
    client = PeerClient("https://upgraded-barnacle-7v4xprj74jwhxr55-5000.app.github.dev/", client_id="user123")
    
    while True:
        target = input("Enter recipient ID: ")
        msg = input("Enter message: ")
        client.send_message(target, msg)
